_posts/2022-04-27-newpipe-0.23.0.md

```diff
@@ -0,0 +1,176 @@
+
+- [@haklc](https://github.com/haklc) added a cool feature where you can change the audio pitch by semitones instead of percentage! This is a nifty tool for musically-inclined users, and also something fun to play with for everyone else. Don't worry, you can toggle a checkbox to revert to the percentage way of doing things.
+
+- [@ktprograms](https://github.com/ktprograms) added a button to the queue menu in the main player which lets you add the current queue to a new or an existing playlist. Incidentally, this can also be used to 'convert' a remote playlist into a local NewPipe playlist, which you can then edit away to your heart's content. Isn't that such a nifty feature?
+
+### Improved
+
+- [@XiangRongLin](https://github.com/XiangRongLin) made the throttling decryption code more resistant to API changes on YouTube's end. This means that there is now a lesser chance of videos slowing down to glacial, virtually unplayable, speeds in the future. Woohoo!
+
+- [@TiA4f8R](https://github.com/TiA4f8R) added some more parameters to the requests the Extractor sends to YouTube, which make them mimic official clients more closely, yielding some benefits like reduced probability of throttling as well as reduced data use.
+
+- [@TiA4f8R](https://github.com/TiA4f8R) also made it so that separate video and audio streams are fetched for livestreams, instead of combined streams. And and and! You can now rewind livestreams back more than 30 seconds! How much you can rewind depends on the livestream in question. It's not a fixed value. Also, please note that rewinding in livestreams can be disabled by individual creators on YouTube.
+
+- [@ktprograms](https://github.com/ktprograms) made the player remember the step size for the speed and pitch adjustment bars in the playback control dialog. Now the dialog doesn't default to the 25% step size each time you open it. Rather, it shows you the last used step size.
+
+- [@Stypox](https://github.com/Stypox) ensured that all the relevant dialogs in the app uniformly show the "OK" and "Cancel" buttons, which are the default on Android, instead of "Yes" and "No" buttons.
+
+- [@Stypox](https://github.com/Stypox) also improved the player UI and navigability for Android TV. This fixed several problems: unreachable buttons in the main player queue, incorrect background theming of the main player menus, and the player controls intermittently showing up below the queue menu. 
+
+- [@SpinHit](https://github.com/SpinHit) added a confirmation dialog when deleting all files in the Download menu.
+
+- [@mauriciocolli](https://github.com/mauriciocolli) noticed that the app used duplicated icons for the night theme, which caused visibility problems for a lot of them. Since the original icons are vector drawables, the duplicated icons were deleted en masse, and instead, color attributes were used to match the original icons to the active theme.
```

Comment:
Here I would also add a note about the fact that some small artifacts have probably been introduced, since this PR affected basically every place in the app where an icon was used. Some have already been reported, see https://github.com/TeamNewPipe/NewPipe/issues/8230#issuecomment-1108703650.

Reply:
```suggestion
- [@mauriciocolli](https://github.com/mauriciocolli) noticed that the app used duplicated icons for the night theme, which caused visibility problems for a lot of them. Since the original icons are vector drawables, the duplicated icons were deleted en masse, and instead, color attributes were used to match the original icons to the active theme. We tried to catch every case where this breaks the icon, but it is likely we missed some. Feel free to report them in the repo (see the last header for more details on that).
```